### DISCUSSION TITLE
How to use prefer_alias?

### DISCUSSION THREAD

**Discussion by danielniccoli** [litestar-org/litestar#2938] [Q&A] [answered]
The changelog shows:

> **Customization of Pydantic integration via PydanticPlugin**

> A new [PydanticPlugin](https://docs.litestar.dev/2/reference/contrib/pydantic.html#litestar.contrib.pydantic.PydanticPlugin) has been added, which can be used to configure Pydantic behaviour. Currently it supports setting a prefer_alias option, which will pass the by_alias=True flag to Pydantic when exporting models, as well as generate schemas accordingly.

How do I configure/enable this? Could someone help out with a code sample?

**Comment by JacobCoffee** [comment, marked as answer]
You can set this by doing something like
```py
pydantic_plugin = PydanticPlugin(prefer_alias=True)
app = Litestar(route_handlers=[...], plugins=[pydantic_plugin])
```

**Reply by JacobCoffee** [reply]
Maybe if this isn't clear we can document it better? Feel free to open an issue on this.. im all for making docs easier to understand

**Reply by JacobCoffee** [reply]
Also @Alc-Alc has a nice MCVE from another issue in [our discord](https://discord.gg/litestar-919193495116337154) that shows real world-ish use:

https://discord.com/channels/919193495116337154/1176079295026839613/1176101578676125696

**Reply by danielniccoli** [reply]
Thanks. And I already have a real-worldish use, which is using snake_case in code, but camelCase in JSON for property names. Pydantic should do that by default in my opinion, but they were not appreciative of my suggestion. Which is why I was super glad when I read that there is now a setting in LiteStar, which works exactly as I hoped for!
